contrib/src/main/java/com/datatorrent/contrib/parser/DelimitedSchema.java

```diff
@@ -0,0 +1,359 @@
+ * "Boolean", "constraints": { "required": "true" } } ] }}
+ */
+public class DelimitedSchema
+{
+
+  /**
+   * JSON key string for separator
+   */
+  private static final String SEPARATOR = "separator";
+  /**
+   * JSON key string for quote character
+   */
+  private static final String QUOTE_CHAR = "quoteChar";
+  /**
+   * JSON key string for line delimiter
+   */
+  private static final String LINE_DELIMITER = "lineDelimiter";
+  /**
+   * JSON key string for fields array
+   */
+  private static final String FIELDS = "fields";
+  /**
+   * JSON key string for name of the field within fields array
+   */
+  private static final String NAME = "name";
+  /**
+   * JSON key string for type of the field within fields array
+   */
+  private static final String TYPE = "type";
+  /**
+   * JSON key string for constraints for each field
+   */
+  private static final String CONSTRAINTS = "constraints";
+  /**
+   * JSON key string for required constraint
+   */
+  public static final String REQUIRED = "required";
+  /**
+   * JSON key string for equals constraint
+   */
+  public static final String EQUALS = "equals";
+  /**
+   * JSON key string for length constraint
+   */
+  public static final String LENGTH = "length";
+  /**
+   * JSON key string for min length constraint
+   */
+  public static final String MIN_LENGTH = "minLength";
+  /**
+   * JSON key string for max length constraint
+   */
+  public static final String MAX_LENGTH = "maxLength";
+  /**
+   * JSON key string for min value constraint
+   */
+  public static final String MIN_VALUE = "minValue";
+  /**
+   * JSON key string for max value constraint
+   */
+  public static final String MAX_VALUE = "maxValue";
+  /**
+   * JSON key string for regex pattern constraint
+   */
+  public static final String REGEX_PATTERN = "pattern";
+  /**
+   * JSON key string for date format constraint
+   */
+  public static final String DATE_FORMAT = "format";
+  /**
+   * JSON key string for locale constraint
+   */
+  public static final String LOCALE = "locale";
+  /**
+   * JSON key string for true value constraint
+   */
+  public static final String TRUE_VALUE = "trueValue";
+  /**
+   * JSON key string for false value constraint
+   */
+  public static final String FALSE_VALUE = "falseValue";
+  /**
+   * delimiter character provided in schema. Default is ,
+   */
+  private int delimiterChar = ',';
+  /**
+   * quote character provided in schema. Default is "
+   */
+  private char quoteChar = '\"';
+  /**
+   * line delimiter character provided in schema. Default is new line character
+   */
+  private String lineDelimiter = "\r\n";
+  /**
+   * This holds the list of field names in the same order as in the schema
+   */
+  private List<String> fieldNames = new LinkedList<String>();
+  /**
+   * This holds list of {@link Field}
+   */
+  private List<Field> fields = new LinkedList<Field>();
+
+  /**
+   * Supported data types
+   */
+  public enum FIELD_TYPE
```

Comment:
Can you please rename this to "FieldType"
All caps is usually used for parameterized classes OR static final constants etc.

I actually got confused in CellProcessorBuilder class when I saw FIELD_TYPE as input param type to be method.